Cargo.toml

```diff
@@ -8,11 +8,17 @@ publish = false
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[profile.release]
+opt-level = "s"
+
 [dependencies]
 
 [dependencies.libc]
 version = "0.2.70"
 
+[dependencies.finder]
+version = "0.1"
+
```

Comment:
Your call here, but I'd suggest to:

1. Make the size optimization a separate commit.
2. Move the "add finder dependency" to the same commit in which you use this `finder` package.

I think this would make each commit more self-contained, and therefore:

1. Easier to understand if we ever need to do Git archaeology in the future.
2. Easier to revert, shall we find some issue.

Reply:
Sure,
1 is fixed in https://github.com/balena-os/takeover/pull/66/commits/1f8f596b8178e0bea2fb4b86f8faed0237244d3f
and 2 is addressed in https://github.com/balena-os/takeover/pull/66/commits/09e0df98384d364402ecd64e088cd2c7bfac9ee7

Was accustomed to committing changes per file/directory, as we usually do in device integration repos :)